src/main/java/org/xbill/DNS/Lookup.java

```diff
@@ -268,7 +268,8 @@ public Lookup(Name name, int type, int dclass) {
       this.searchPath = getDefaultSearchPath();
       this.cache = getDefaultCache(dclass);
     }
-    this.ndots = defaultNdots;
+    // negative defaultsNdots means that the config provider did not support the config type
+    this.ndots = defaultNdots == -1 ? 1 : defaultNdots;
```

Comment:
Very good finding! I wonder though, maybe it would be better to default to 1 instead of -1 in the `ResolverConfig` interface, especially if we would default to 1 here anyway.

And something else: the tests set `CONFIGPROVIDER_SKIP_INIT`, so why was the `WindowsResolverConfig` used anyway?

Reply:
I would be happy to simplify `ResolverConfigProvider` to return the default if the value is not set.

I believe that `CONFIGPROVIDER_SKIP_INIT` fails to prevent WindowsResolverConfigProvider due to the fact that some other test triggers it before `LookupTest` is run. I have verified that the before() and after() methods gets properly run with a debugger